xapian-core/include/xapian/weight.h

```diff
@@ -463,9 +463,107 @@ class XAPIAN_VISIBILITY_DEFAULT BoolWeight : public Weight {
 
 /// Xapian::Weight subclass implementing the tf-idf weighting scheme.
 class XAPIAN_VISIBILITY_DEFAULT TfIdfWeight : public Weight {
-    /* Three character string indicating the normalizations for tf(wdf), idf and
-       tfidf weight. */
-    std::string normalizations;
+    /* These three enum classes should use integers only up to 255. Otherwise
+       during serialise(), the value may get truncated by static_cast<> and
+       the desired result will not be achieved. */
```

Comment:
Rather than just documenting this in the comment (which could be missed), I think it makes more sense to instead specify the underlying type of the `enum` as `unsigned char`.  Then the compiler knows to complain if an out of range value is specified - e.g. GCC gives the error:

```
error: enumerator value ‘256’ is outside the range of underlying type ‘unsigned char’
```

Better to let the compiler check such requirements for us when we can get it to.

Reply:
Yes, This is better. Updated.